extensions/compose/src/extension.ts

```diff
@@ -138,16 +145,45 @@ export async function activate(extensionContext: extensionApi.ExtensionContext):
   // Need to "ADD" a provider so we can actually press the button!
   // We set this to "unknown" so it does not appear on the dashboard (we only want it in preferences).
   const providerOptions: extensionApi.ProviderOptions = {
-    name: 'Compose',
-    id: 'Compose',
+    name: composeDisplayName,
+    id: composeDisplayName,
     status: 'unknown',
     images: {
-      icon: './icon.png',
+      icon: imageLocation,
     },
   };
 
-  providerOptions.emptyConnectionMarkdownDescription = `Compose is a specification for defining and running multi-container applications. We support both [podman compose](https://docs.podman.io/en/latest/markdown/podman-compose.1.html) and [docker compose](https://github.com/docker/compose) commands.\n\nMore information: [compose-spec.io](https://compose-spec.io/)`;
+  providerOptions.emptyConnectionMarkdownDescription = composeDescription;
 
   const provider = extensionApi.provider.createProvider(providerOptions);
   extensionContext.subscriptions.push(provider);
+
+  // The location of the binary (local storage folder)
+  const binaryPath = path.join(
+    extensionContext.storagePath,
+    'bin',
+    os.isWindows() ? composeCliName + '.exe' : composeCliName,
+  );
+  let binaryVersion = '';
+
+  // Use getCliVersion to try and get the version of the binary.
+  // If this fails, return output to console the error.
+  try {
+    binaryVersion = await getCliVersion(binaryPath, '--version');
```

Comment:
Version detection should not be a part of extension activation function. All long running tasks should not be slowing down extension activation.

Reply:
For `docker-compose` cli `version --short` options should be used, so we do not need to parse output and just use `stdout.trim()` as a version.
```
eskimo@degolovi-mac xterm.js % docker-compose version --short
2.20.1
eskimo@degolovi-mac xterm.js %
```

Reply:
Do you have a recommendation where we should have it separate from activate? 

Just looking through some other extensions (kind, podman) and I see that they also call the CLI before activating.

Reply:
Maybe add a callback in the podman desktop API to go through and "activate" each CLITool if the provider has it?

Reply:
Thanks @jeffmaury I'll modify the code that we update it via postActivate instead.

Reply:
Updated and moved to postActivation.